tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -300,6 +301,44 @@ private:
    std::unordered_map<DescriptorId_t, RClusterDescriptor> fClusterDescriptors;
 
 public:
+   class RFieldDescriptorRange {
+   private:
+      const RFieldDescriptor& fField;
+   public:
+      class RIterator {
+      private:
+         const std::vector<DescriptorId_t>& fFieldChildren;
+         std::size_t fIndex = kInvalidNTupleIndex;
+      public:
+         using iterator_category = std::forward_iterator_tag;
```

Comment:
I avoided inheriting from `std::iterator` here since it was deprecated in C++17 .